src/scheduler/server_info.cpp

```diff
@@ -2958,7 +2836,48 @@ counts_max(counts *cmax, counts *ncounts)
 			}
 		}
 	}
-	return cmax_head;
+}
+
+// overloaded 
+void
+counts_max(counts_map &cmax, counts *ncounts)
```

Comment:
There is a lot of code between these two functions, any chance some of it can be moved into a helper function to be called by both overloads?